### ISSUE TITLE
*Reference scope bindings are not visible to Light DOM if no template is defined

### ISSUE THREAD

**Issue by dylanrtt** [canjs/canjs#1854]
If I do the following:

```html
outer scope: {{parentAttr}}

<my-component #parent-attr="{childAttr}">
  light DOM: {{parentAttr}}
</my-component>
```

http://jsbin.com/zajiloqaze/edit?html,js,output

The outer scope can see the value of ```parentAttr``` but the light DOM (aka user content) of ```my-component``` cannot. With ```leakScope:true```, you could just read directly from ```childAttr``` but it's not clear where that is coming from so being able to do the above could be useful.

I like the idea of being able to explicitly expose a component's view model to the outer context primarily because it allows you to use properties/methods of the component's viewmodel even when its defined with ```leakScope:false``` which I prefer to always do because:

1) it forces data to be passed in explicitly which increases clarity
2) it prevents random junk from leaking into the component's template or light DOM

At the very least, I think this should work with ```leakScope:false```. I originally expected it to  because, in theory, the light DOM would be reading directly from the outer context but I was unpleasantly surprised :(

**Comment by justinbmeyer** [comment]
I thought this was tested. I think it should work the way you describe.

**Comment by dylanrtt** [comment]
I found the issue. It's happening because no template is defined for the component. If I put ```template:can.stache('<content/>')``` on the component definition it fixes it.

Lacking a template also affects lexical scoping and renders ```leakScope:false``` useless whereas it should probably still prevent the light DOM from reading from the component's viewmodel. Even before #reference values, you could still have a component with no template that used a view model that wasn't used by the light DOM.

**Comment by dylanrtt** [comment]
Seems to be fixed in minor.

**Comment by dylanrtt** [comment]
I was mistaken. This is still an issue and I just didn't update the syntax in the jsbin correctly to test it.

```
Outer Scope: {{*bar}}
<my-component {^foo}="*bar">
  Light DOM: {{*bar}}
</my-component>
```

http://jsbin.com/zegotuwope/edit?html,js,output

It's also an issue in examples like this (`bar` changing does not update `foo`):

```
<foo-bar {(foo)}="*shared">
  <fiz-biz {(bar)}="*shared"></fiz-biz>
</foo-bar>
```

Still fixable by adding a template, even `template: can.stache('<content/>')`.

**Comment by justinbmeyer** [comment]
I thought this should fix it: https://github.com/canjs/canjs/issues/2029

Do you have 2.3.1?

**Comment by dylanrtt** [comment]
Ah, it is fixed in `2.3.1`. I usually just test in jsbin with minor and assumed that would be up to date. Next time I'll use a specific version.

I'll upgrade now. Thanks!

**Comment by justinbmeyer** [comment]
minor is not being used right now.  master is where everything goes until probably we start putting things in the major (3.0) branch.
